app/models/ability.rb

```diff
@@ -35,7 +35,41 @@ def initialize(user)
     #   end
 
     user ||= User.new # guest user (not logged in)
+    id = user.id
+
     # All users can only update their own user attributes
-    can :update, User, id: user.id
+    alias_action :create, :read, :update, :destroy, :to => :crud
+
+    can :crud, Team
+    can :update, User, id: id
+
+    can :assign_ownership, Team, Team do |team|
+      team.owners.include? user
+    end
+
+    can :delete_ownership, Team, Team do |team|
+      team.owners.include? user
+      team.owners.length > 1
+    end
+
+    can :delete_membership, Team, Team do |team|
+      team.owners.include? user and Ability.number_of_owners_after_delete(team, team_member) > 0
+    end
+
+    can :delete_membership, Team, Team do |team|
+      Integer(id) == team_member and Ability.number_of_owners_after_delete(team, team_member) > 0
```

Comment:
I think you accidentally removed the Integer conversion from `team_member` (where it ist necessary) and left it on `id` (where it can be omitted).